collection_test.go

```diff
@@ -217,3 +217,24 @@ func TestPush(t *testing.T) {
 		t.Error("The keys were messed up :/")
 	}
 }
+
+func TestMerge(t *testing.T) {
```

Comment:
I think it'd be interesting to use at least one `CollectMap` here, to ensure the keys are preserved.